src/hoc/BadgerBase/BadgerBase.js

```diff
@@ -283,6 +308,56 @@ const BadgerBase = (Wrapped: React.AbstractComponent<any>) => {
 			this.setState({ intervalUnconfirmed: intervalUnconfirmedNext });
 		};
 
+		setupWatchInvoice = async () => {
+			const { paymentRequestUrl } = this.props;
+
+			this.ws = new WebSocket(
+				`wss://pay.bitcoin.com/s/${paymentRequestUrl.slice(26)}`
+			);
+
+			this.ws.onmessage = (evt) => {
+				// listen to data sent from the websocket server
+				const initialInvoiceInfo = JSON.parse(evt.data);
+				const invoiceStatus = initialInvoiceInfo.status;
+				if (invoiceStatus === 'paid') {
+					return this.paymentSendSuccess();
+				}
+				if (invoiceStatus === 'expired') {
+					return this.invoiceExpired();
+				}
+
+				// If invoice is not expired or paid, start the timer (add this logic after timer works) TODO
+				// Get current UTC time for timer
+				const now = new Date();
```

Comment:
Can be shortened to 
`const nowUTC = Date.now()`

Reply:
be00b10